NARFO_BE/NARFO_BE/Controllers/MembersController.cs

```diff
@@ -2,121 +2,133 @@
+                newMember.SURNAME = "admin";
                 newMember.Username = "admin";
-                newMember.FirstName = "admin";
-                newMember.Password = "admin";
+                newMember.Firstname = "admin";
+                newMember.Email = "admin@admin.com";
+               
                 _context.Members.AddAsync(newMember);
                 _context.SaveChanges();
             }
         }
-        
-        public void ListofMembers(List<Members> members)
-        {
-            this.members = members;
-        }
-        
 
-        // GET: api/Members
-        [HttpGet]
-        public async Task<ActionResult<IEnumerable<Members>>> GetMembers()
-        {
-            return await _context.Members.ToListAsync();
-        }
-        public IEnumerable<Members> GetAllMembers()
+        public void ListofMembers(List<_Member> members)
         {
-            return members;
+            this.members = members;
         }
 
-        // GET: api/Members/5
-        [HttpGet("{id}")]
-        public async Task<ActionResult<Members>> GetMembers(int id)
+        //
+        // POST: /Account/Login
+        [HttpPost("get/login")]
+        public async Task<ActionResult <_Member>> Login([FromBody] _Member model)
         {
-            var members = await _context.Members.FindAsync(id);
+            
+            _Member user = await _context.Members.FirstOrDefaultAsync(member=>member.Email == model.Email && member.Password == encryption.ComputeHash(model.Password));
 
-            if (members == null)
+            if (user == null)
             {
-                return NotFound();
+                return BadRequest(new { status = "Failed", message = "Invalid login" });
             }
+            else {
 
-            return members;
+                return Ok(new { status = "Success", member = model });
+            }
+            
+            
         }
 
-        // PUT: api/Members/5
-        [HttpPut("{id}")]
-        public async Task<IActionResult> PutMembers(int id, Members members)
+        private ActionResult<_Member> Json(object p)
         {
-            if (id != members.Id)
-            {
-                return BadRequest();
-            }
+            throw new NotImplementedException();
+        }
 
-            _context.Entry(members).State = EntityState.Modified;
 
-            try
+        // GET: Member/Email
+        [HttpGet("get/all/user")]
+        public  async Task<ActionResult<MemberPrototype>> GetMembersEmail()
+        {
+           
+           List<MemberPrototype> endpoint = new List<MemberPrototype>();
+            foreach (_Member d in  await _context.Members.ToArrayAsync())
             {
-                await _context.SaveChangesAsync();
+                if(d.Username !=  null && d.Email != null )
+            endpoint.Add(new MemberPrototype(d.Username,d.Email));       
             }
-            catch (DbUpdateConcurrencyException)
+            if(endpoint == null)
```

Comment:
endpoint will never be null since you assigned it to be a "new List<MemberPrototype>()" on line 75.

Reply:
It's contents may be null though?

Reply:
true..  I'll change that to endpoint.length(or size) < 0 ..  **is
 that better?????**

Reply:
You can also use `if(!endpoint.Any())`

Reply:
You cant use length on a `List`, rather use `endpoint.Any()`

Reply:
thank you